packages/prerender-fargate/lib/prerender-fargate.ts

```diff
@@ -0,0 +1,100 @@
+import { Construct } from 'constructs';
+import * as ecs from 'aws-cdk-lib/aws-ecs';
+import * as ecsPatterns from 'aws-cdk-lib/aws-ecs-patterns';
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+import { Certificate } from 'aws-cdk-lib/aws-certificatemanager';
+import { HostedZone } from 'aws-cdk-lib/aws-route53';
+import { Bucket, BlockPublicAccess } from 'aws-cdk-lib/aws-s3'
+import * as ecrAssets from 'aws-cdk-lib/aws-ecr-assets';
+import { AccessKey, User } from 'aws-cdk-lib/aws-iam';
+import { Duration, RemovalPolicy, Stack } from 'aws-cdk-lib';
+import * as path from 'path';
+
+export interface PrerenderOptions {
+    prerenderName: string,
+    domainName: string,
+    vpcId: string,
+    bucketName?: string,
+    expirationDays?: number,
+    basicAuthList: Array<string[]>,
+    certificateArn: string,
+}
+
+export class PrerenderFargate extends Construct {
+    constructor(scope: Construct, id: string, props: PrerenderOptions) {
+        super(scope, id);
+
+        // Create bucket for prerender storage
+        const bucket = new Bucket(this, `${props.prerenderName}-bucket`, {
+            bucketName: props.bucketName,
+            lifecycleRules: [{
+                enabled: true,
+                expiration: Duration.days(props.expirationDays || 7) // Default to 7 day expiration
+            }],
+            removalPolicy: RemovalPolicy.DESTROY,
+            autoDeleteObjects: true,
+            blockPublicAccess: BlockPublicAccess.BLOCK_ALL,
+        });
+
+        // Configure access to the bucket for the container
+        const user = new User(this, 'PrerenderAccess');
+        bucket.grantReadWrite(user);
+
+        const accessKey = new AccessKey(this, 'PrerenderAccessKey', {
+            user: user,
+            serial: 1
+        });
+
+        const vpc = ec2.Vpc.fromLookup(this, "vpc", { vpcId: props.vpcId });
+
+        const cluster = new ecs.Cluster(this, `${props.prerenderName}-cluster`, { vpc: vpc });
+
+        const directory = path.join(__dirname, 'prerender');
+        const asset = new ecrAssets.DockerImageAsset(this, `${props.prerenderName}-image`, {
+            directory,
+        });
+
+        // Create a load-balanced Fargate service 
+        const fargateService = new ecsPatterns.ApplicationLoadBalancedFargateService(
+            this,
+            `${props.prerenderName}-service`,
+            {
+                cluster,
+                serviceName: `${props.prerenderName}-service`,
+                desiredCount: 1,
+                cpu: 512, // 0.5 vCPU (this may need to be increased)
```

Comment:
How about if we pass, resource limits as props. Rather than hard-coding it?

Reply:
Good idea. I might add these as defaults but allow a manual override :+1: